### PR TITLE
Chore: Allow self-serve for ssoSettingsApi feature toggle

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**

1. Allow self-serve for `ssoSettingsApi`.
2. Align feature toggle validations with the changes of https://github.com/grafana/grafana/pull/81562 (AllowSelfServe for PublicPreview)

**Why do we need this feature?**

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/featuremgmt/toggles_gen_test.go [hunk 1]
```diff
@@ -49,8 +49,8 @@ func TestFeatureToggleFiles(t *testing.T) {
 			if flag.Name != strings.TrimSpace(flag.Name) {
 				t.Errorf("flag Name should not start/end with spaces.  See: %s", flag.Name)
 			}
-			if flag.AllowSelfServe && flag.Stage != FeatureStageGeneralAvailability {
-				t.Errorf("only allow self-serving GA toggles")
+			if flag.AllowSelfServe && !(flag.Stage == FeatureStageGeneralAvailability || flag.Stage == FeatureStagePublicPreview || flag.Stage == FeatureStageDeprecated) {
```
Comment (mgyongyosi) [inline pkg/services/featuremgmt/toggles_gen_test.go:52]: @jcalisto Could you please take a look at this? I aligned this to the changes of https://github.com/grafana/grafana/pull/81562.

Comment (jcalisto) [reply pkg/services/featuremgmt/toggles_gen_test.go:52]: LGTM, thanks for updating this check @mgyongyosi!

**Review by jcalisto — APPROVED** [review]

**Pull request merged**